controllers/snapshot/snapshot_adapter.go

```diff
@@ -42,6 +41,39 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	ReasonEnvironmentNotInNamespace = "EnvironmentNotInNamespace"
+	ReasonUnknownError              = "UnknownError"
+)
+
+type IntegrationError struct {
```

Comment:
I know we decided against putting the error definition in `pkg`, but we could instead place it in the `helpers` package so it can be available to other controllers, wdyt?

Reply:
Errors is moved to errorhandlers.go of helpers pkg. Thanks.

Reply:
Hold on, need some update.

Reply:
Ready to review.